### ISSUE TITLE
"package io/fs is not in GOROOT" on Go 1.15

### ISSUE THREAD

**Issue by zikaeroh** [golang-migrate/migrate#480]
**Describe the Bug**

golang-migrate no longer builds in Go 1.15 due to the addition of `io/fs`, imported via tests.

**Steps to Reproduce**

Add v4.14.0 to the project, then run `go mod tidy`. I get:

```
github.com/hortbot/hortbot/internal/db/migrations imports
	github.com/golang-migrate/migrate/v4/database/postgres tested by
	github.com/golang-migrate/migrate/v4/database/postgres.test imports
	github.com/golang-migrate/migrate/v4/source/file imports
	github.com/golang-migrate/migrate/v4/source/iofs imports
	io/fs: package io/fs is not in GOROOT (/usr/lib/go/src/io/fs)
github.com/hortbot/hortbot/internal/db/migrations imports
	github.com/golang-migrate/migrate/v4/database/postgres tested by
	github.com/golang-migrate/migrate/v4/database/postgres.test imports
	github.com/golang-migrate/migrate/v4/source/file imports
	github.com/golang-migrate/migrate/v4/source/iofs tested by
	github.com/golang-migrate/migrate/v4/source/iofs.test imports
	embed: package embed is not in GOROOT (/usr/lib/go/src/embed)
```

**Expected Behavior**

No output.

**Migrate Version**

v4.14.0

**Loaded Source Drivers**

N/A

**Loaded Database Drivers**

N/A

**Go Version**

go version go1.15.5 linux/amd64

**Stacktrace**

N/A

**Additional context**

N/A

**Comment by dhui** [comment]
Looks like [modules ignore build constraints](https://golang.org/ref/mod#resolve-pkg-mod)
For now use v4.13.0
I'll release v4.14.1 later that removes the iofs source driver until Go 1.16 is the minimum supported version by `migrate`

**Comment by zikaeroh** [comment]
I don't think that's the whole issue. If you just clone this repo and run `go mod tidy`, it fails.

```
github.com/golang-migrate/migrate/v4/source/iofs imports
	io/fs: package io/fs is not in GOROOT (/usr/lib/go/src/io/fs)
github.com/golang-migrate/migrate/v4/source/iofs tested by
	github.com/golang-migrate/migrate/v4/source/iofs.test imports
	embed: package embed is not in GOROOT (/usr/lib/go/src/embed)
```

**Comment by zikaeroh** [comment]
Unless I'm mistaken, the issue is that the `iofs` package is not buildable in Go 1.15. There are build tags in the `file` source, but the `iofs` package needs build tags too, even if they just replace things with noops or panics.

**Comment by wayneashleyberry** [comment]
I'm seeing the same issue in a project of mine, both `go test` and `go build` work but `go mod tidy` fails.

```
github.com/overhq/over-projects-api/pkg/migrator imports
	github.com/golang-migrate/migrate/v4/database/mysql tested by
	github.com/golang-migrate/migrate/v4/database/mysql.test imports
	github.com/golang-migrate/migrate/v4/source/file imports
	github.com/golang-migrate/migrate/v4/source/iofs imports
	io/fs: package io/fs is not in GOROOT (/usr/local/Cellar/go/1.15.5/libexec/src/io/fs)
github.com/overhq/over-projects-api/pkg/migrator imports
	github.com/golang-migrate/migrate/v4/database/mysql tested by
	github.com/golang-migrate/migrate/v4/database/mysql.test imports
	github.com/golang-migrate/migrate/v4/source/file imports
	github.com/golang-migrate/migrate/v4/source/iofs tested by
	github.com/golang-migrate/migrate/v4/source/iofs.test imports
	embed: package embed is not in GOROOT (/usr/local/Cellar/go/1.15.5/libexec/src/embed)
```

**Comment by ksegun** [comment]
We also make use of this library and this is breaking our builds because of the module dependencies we have to pin over 30+ repos to v4.13.0 is there an outlook for when v4.14.1 will be available?

**Comment by johejo** [comment]
I'm the author of the #472. 
Sorry for breaking the builds of many users.
I made a Pull Request for revert.
Please wait for the next release by maintainer.

**Comment by zikaeroh** [comment]
Regardless of the revert, when the package is reintroduced (which I would still do for 1.16), it will work with the build tags guarding the imports of `io/fs` itself. You probably don't need a full revert if you stick `// +build go1.16` into the existing files in `iofs` and add a single file with:

```
// +build !go.1.16

package iofs
```

That will guard its use in versions below 1.16.

**Comment by zikaeroh** [comment]
Hm, I say that, but you've done that in `iofs` already...

**Comment by zikaeroh** [comment]
Ugh, this is golang/go#40067. Sorry for the noise.

I guess this driver could live in another repo until 1.16 is the minimum. Going via `http.FileSystem` works as well, though is clearly an unnecessary dep.

**Comment by ksegun** [comment]
> I'm the author of the #472.
> Sorry for breaking the builds of many users.
> I made a Pull Request for revert.
> Please wait for the next release by maintainer.

Thank you so much, I know it was not your intent, and thank you for this library.

**Comment by dhui** [comment]
Fixed in [v4.14.1](https://github.com/golang-migrate/migrate/releases/tag/v4.14.1)
